### ISSUE TITLE
Remove BackBlast name from scripts

### ISSUE THREAD

**Issue by LeeBergstrand** [LeeBergstrand/BackBLAST_Reciprocal_BLAST#45]
![Screen Shot 2019-08-31 at 1 48 00 PM](https://user-images.githubusercontent.com/5819462/64069024-18348800-cbf6-11e9-9978-09daebf9aba4.png)

We can remove the "backblast" from infront of all these script's names.

**Comment by jmtsuji** [comment]
I gave the support scripts the "BackBLAST" filename prefix for namespace
purposes. Imagine a user installing BackBLAST. They will have the main
"BackBLAST" (shell script) executable plus all of the support scripts added
to the PATH. I think "BackBLAST_generate_heatmap.R" is preferable to
"generate_heatmap.R" for clarity.

An alternative would be to not add the support scripts to the PATH but call
them a different way. For example, the scripts could be called through the
main "BackBLAST" executable fairly easily. E.g., "BackBLAST heatmap" could
call the "generate_heatmap.R" script.

Let me know what your preference is.

**Comment by jmtsuji** [comment]
@LeeBergstrand nudge

**Comment by LeeBergstrand** [comment]
@jmtsuji I would generally avoid cluttering up the path when possible. However, I don't know what the standard operating procedure is with Conda. Does each environment get its own path that activates when the environment is activated? Will have the script calling the other scripts to break if the pathing changes?

**Comment by LeeBergstrand** [comment]
@jmtsuji I'm generally leaning towards this I can.

> An alternative would be to not add the support scripts to the PATH but call
them a different way. For example, the scripts could be called through the
main "BackBLAST" executable fairly easily. E.g., "BackBLAST heatmap" could
call the "generate_heatmap.R" script.

**Comment by LeeBergstrand** [comment]
What does atlas do for its scripts?

**Comment by jmtsuji** [comment]
> @jmtsuji I would generally avoid cluttering up the path when possible. However, I don't know what the standard operating procedure is with Conda. Does each environment get its own path that activates when the environment is activated? Will have the script calling the other scripts to break if the pathing changes?

I think it will work to have one script call the other scripts. I'm thinking specifically of having the support scripts in a folder (e.g., `BackBLAST-utils`) in the same folder as the main `BackBLAST` script. `BackBLAST` can then look in the same folder where it is located to find the utils folder, by default. Or the user can specify via a flag where the utils are located. I don't think this will cause an issue with conda envs.

Based on a quick glance, I found two different methods ATLAS uses to work with its custom scripts:
- Installs them as part of the ATLAS python package, then load the functions from them, e.g., `from utils.parsers_bbmap import parse_comments` in [`qc.snakefile`](https://github.com/metagenome-atlas/atlas/blob/master/atlas/rules/qc.snakefile)
- Uses them via the [Snakemake `script` command](https://snakemake.readthedocs.io/en/stable/snakefiles/rules.html#external-scripts). Inputs from Snakemake are passed as objects directly into the script with defined naming, and the script uses them. Means that the script needs to have an option to directly receive Snakemake objects, e.g., [`parse_cat_taxonomy.py`](https://github.com/metagenome-atlas/atlas/blob/master/atlas/scripts/parse_cat_taxonomy.py). Not Bash compatiable.

Neither of these options works well with Bash. So I wonder if my proposed solution of having one script call all the others is most practical?

For now, I've merged the current PR (#44) that you approved. I can deal with the pathing of scripts in an upcoming PR based on what we decide here.
